bop_toolkit_lib/score.py

```diff
@@ -177,158 +177,135 @@ def calc_pose_detection_scores(
     :param obj_ids: ID's of considered objects.
-    :param matches: Info about matching pose estimates to ground-truth poses
+    :param matches: List of per-GT dictionary with info about GT-to-detection matching.
       (see pose_matching.py for details).
-    :param scene_errs: List of dictionaries with:
+    :param errs: List of per-detection dictionary with the following items:
       - 'im_id': Image ID.
       - 'obj_id': Object ID.
       - 'est_id': ID of the pose estimate.
       - 'score': Confidence score of the pose estimate.
       - 'errors': Dictionary mapping ground-truth ID's to errors of the pose
           estimate w.r.t. the ground-truth poses.
-    :param n_top: Number of top pose estimates to consider per test target.
-    :param visib_gt_min: Min visiblity for GT. Default: 0.1
-    :param ignore_object_visible_less_than_visib_gt_min: Whether ignore objects visible less than visib_gt_min. Default: True
+    :param visib_gt_min: GT annotations visible from less than this threshold are ignored.
+        from less than visib_gt_min.
     :param do_print: Whether to print the scores to the standard output.
-    :return: Dictionary with the evaluation scores.
+    :return: Dictionary with the per-object AP scores.
     """
+
     # Count the number of visible object instances in each image.
-    insts = {i: {j: defaultdict(lambda: 0) for j in scene_ids} for i in obj_ids}
-    for m in matches:
-        if m["valid"]:
-            insts[m["obj_id"]][m["scene_id"]][m["im_id"]] += 1
+    insts = {obj_id: {scene_id: defaultdict(lambda: 0) for scene_id in scene_ids} for obj_id in obj_ids}
+    for match in matches:
+        insts[match["obj_id"]][match["scene_id"]][match["im_id"]] += 1
 
-    tars = 0  # Total number of targets.
-    obj_tars = {i: 0 for i in obj_ids}  # Targets per object.
-    scene_tars = {i: 0 for i in scene_ids}  # Targets per scene.
-    for obj_id, obj_insts in insts.items():
-        for scene_id, scene_insts in obj_insts.items():
-            # Count the number of targets for the current object in the current scene.
-            if n_top > 0:
-                count = sum(np.minimum(n_top, list(scene_insts.values())))
-            else:
-                count = sum(list(scene_insts.values()))
+    # Calculate per-object detection scores.
+    scores_per_object = {}
+    num_targets_per_object = {}
+    for obj_id in obj_ids:
 
-            tars += count
-            obj_tars[obj_id] += count
-            scene_tars[scene_id] += count
+        # Per-GT info about GT-to-detection matching (`len(obj_matches)` is the number
+        # of GT annotations of the current object).
+        obj_matches = [match for match in matches if match["obj_id"] == obj_id]
 
-    # For each object, sort the pose estimates by confidence score.
-    # Then calculate the average precision for each object.
-    scores_per_object = {}
-    num_instances_per_object = {}
-    total_num_false_ignore = 0
```

Comment:
Could you please add `total_num_gts_ignored` of all objects so we can check it easily with unit tests?